### PR TITLE
add solution

### PR DESCRIPTION
**Pull request by ssaviak**

### CODE REVIEW

**Review by l0ad1nggg — APPROVED** [review]
Well done! Great job :)

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const ACTION_CLEAR = 'clear';
+  const ACTION_ADD = 'addProperties';
+  const ACTION_REMOVE = 'removeProperties';
```
Comment (SerhiiKirik) [inline src/transformStateWithClones.js:12]: You can combine it in object
```
const actions = {
  clear: 'clear'
}

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -7,7 +7,34 @@
...
+  const stateClone = Object.assign({}, state);
+  const states = [];
+
+  for (const action of actions) {
+    if (action.type === ACTION_CLEAR) {
```
Comment (SerhiiKirik) [inline src/transformStateWithClones.js:17]: just do it with `switch / case`, it would be better

File: src/transformStateWithClones.js [hunk 3]
```diff
@@ -7,7 +7,34 @@
...
+      for (const key in stateClone) {
+        delete stateClone[key];
+      }
+      states.push(Object.assign({}, stateClone));
```
Comment (SerhiiKirik) [inline src/transformStateWithClones.js:21]: you can just do it 1 time after your checks
```suggestion
```

**Review by SerhiiKirik — CHANGES_REQUESTED** [review]
Great 🔥 
Let's improve your solution 🧑‍💻

**Review by DanilWeda — APPROVED** [review]
Nice job.
Approved.

**Pull request closed without merging**